Merging mossland-developmental with master

Esqatomsmashers-Students opened the pull request:
mossland-developmental is currently in a working condition and ready to be merged with master. Soon we are looking to make branches related to specific subsystems.

Mossk2868 committed:
Got rid of questionable deadzoning. Re-organized subsystems. Updated commands for moving to automated positions. Added possible limit switch code for automatic encoder resetting. Added logic so that the linear slide can go further out when the wrist is down. Added toggle for Bosco's specimen arm claw. NOT YET TESTED! Would like to test and use for competition though.

Mohmed1222 committed:
Added todo comment about finite states.

Mohmed1222 committed:
Added new/old controls.

Bosco-maker committed:
Merge branch 'FIRST-Tech-Challenge:master' into moss-testing

Mossk2868 committed:
Not tested - Removed comments, removed features for testing

- Removed old and confusing comments in the teleop class
- Removed passing in the SpinningWristSubsyste object in the ArmSubsystem instanciation for testing reasons
    - I suspect that some of the issues caused during previous testing were caused by the system for limitting the arm position based on the wrist position
    - For future testing we should also try removing the wrist moving functions in the CommandManager.

Commit:
TESTED WORKS - Fixed auto move to position

- Replaced run commands in auto to position sequential command groups to custom move wrist commands in the wrist subsystem
- We did this because the commands get caught on the run command and would not proceed

Commit:
Tested Works - Changed the order of auto to position commands and added the variable slide max-lengths

- Added made it so that the wrist only goes in then out when it is already out
- Added the code so that the slide could go further when the wrist was not in outtake position

Commit:
Tested Working - Added logic for variable linear max extension
- Added all of the logic in spinning wrist subsystem for variable linear max extension
- removed old logic for this

Mossk2868 committed:
Not Tested - Fixed logic for variable linear max extension lengths in SpinningWristSubsystem periodic function

 - Made it so that the wrist would set the position of the slide to its current position when it needs to retract rather than its max position
 - This is to fix the issue where when the max extension should have been much higher the wrist subsystem was setting it to low without the possibility for it to change
 - Not sure if this works as it has not been tested, but I am fairly confident!

Bosco-maker committed:
Not works: fix fieldCentric drive with RR

Bosco tried to figured out what did not work and it was the basic fieldCentric drive logic.

the logic was fixed but the control was off

Bosco-maker committed:
Merge remote-tracking branch 'origin/mossland-developmental' into mossland-developmental

Bosco-maker committed:
Working: robot and field centric

Found that the strafe and drive input for RR was wrong.
Dread and happiness felt as I have to move on to the sample arm

Bosco-maker committed:
Semi-worked: added setArmMaxPowerCommand

I encountered a delay problem after cycling the arm for 3-4 times.
Using a separete did not solve the problem by test.

Esqatomsmashers-Students committed:
Finalized some changes with the variable arm extension limits and tested everything

All is working
 - Changed the variable max length logic for the arm so that the wrist can be operated when the arm is extended in the upright position
 - Tested the arm and field centric driving - everything works
 - The pid and feedforward values need to be tuned but it's not that urgent

In the future I think we should also make the SpinningWristSubsystem a subclass of the armsubsystem because they are currently very interconnected

Esqatomsmashers-Students committed:
Works: fieldCentric telemetry and tuned specimen position

Mossk2868 committed:
Merge branch 'master' into mossland-developmental

Mossk2868 committed:
Revert "Merge branch 'master' into mossland-developmental"

This reverts commit 7549ebf3f6723e15a23587f5c4b987eeb1258507, reversing
changes made to d233293614cb0aa9429f641b7f686dd90d5b24ba.

Mossk2868 reviewed (approved):
Everything looks good to me, I think it's ready to be merged!

Bosco-maker commented:
It was tested and it works, approved!

Esqatomsmashers-Students committed:
Tested: RR completely tuned

Bosco-maker reviewed (approved):
Genius! I approve this change( finally figured out how to approve PR)

Esqatomsmashers-Students committed:
To be further tested: LeftBluePushAuto

Esqatomsmashers-Students committed:
To be further tested: RightBlueSpecimenAuto

it can reliably score the first specimen and retrieve the first sample, other functions have to be tested

Esqatomsmashers-Students committed:
Working: LED subsystem

It just has the capability to program an LED. Better formatting needed

Esqatomsmashers-Students committed:
Merge branch 'drivebase-developmental' into mossland-developmental

Esqatomsmashers-Students committed:
Changed driving controls once again!

 - Made it so that the speed variation trigger is fast by default but can still be slow too (a request by Cheryl)

Esqatomsmashers-Students committed:
Added one last thing to fix driver controls - TESTED WORKS

The robot was starting at 50% speed and then whenever the speed variation trigger was used it was reset to one. Just removed what turned it to 50% so it starts at 100%.

BoscoTU committed:
To be tested: goToBasket button

Driver can press start button to automatically drive the robot to where it started in teleop

Esqatomsmashers-Students committed:
Changed the names of classes

 - Changed the names of classes and make a couple of fixes here and there!

Bosco-maker committed:
To be tested: bring back RightPushAuto

Esqatomsmashers-Students committed:
Works: Renamed Autos

Esqatomsmashers-Students committed:
Works:Place Autos into a separate file

Bosco-maker committed:
To be tested: rework on RightSpecimenAuto

It acquires all the samples at once than start scoring specimens, popular team strats!!
Hope it works

Mossk2868 committed:
Renamed packages in accordance with proper naming convention

Some of the packages had capital letters in their names which I removed.

Mossk2868 committed:
Added code for automatic encoder resetting with limit switches

 - Uncommented most of the previous code for automatic slide limit switch resetting
 - Added the hardwaremap name for the slide limit switch found on the flotsam wiring diagram
 - Added in commented code for automatic arm encoder resetting (very similar to that of the linear slide)
     - The reason for leaving it commented out is because I don't yet know the name as it is not in the wiring diagram -> please add the name and uncomment before testing
 - The way it works is the armSubsystem constantly checks to see if either of the limit switches are pressed down and resets the encoders accordingly
 - These changes are NOT YET TESTED!!!!

The overall reason for adding this was because now and then the encoder values become wrong and it is somewhat difficult for the drivers to do this manually. This is meant as a driver enhancement.

Esqatomsmashers-Students committed:
To be tested:changed LED power source

Esqatomsmashers-Students committed:
To be tested: Added LeftSampleAuto

Esqatomsmashers-Students committed:
To be tested: Changed opmode to OpMode

Esqatomsmashers-Students committed:
Works: limit switches debugging

Esqatomsmashers-Students committed:
to be tested: SpecimenArmSubsystem adjustment

Esqatomsmashers-Students committed:
to be tested: Second SpecimenArmSubsystem adjustment

Esqatomsmashers-Students committed:
Not works: adjust RightSpecimenAuto

Esqatomsmashers-Students committed:
Works: using sample arm in auto through commands

Esqatomsmashers-Students committed:
Not working: SpecimenAuto

Esqatomsmashers-Students committed:
Semi-working: SpecimenAuto

Bosco-maker committed:
To be tested: adjusted action complete logic

Esqatomsmashers-Students committed:
Works: SpecimenArmSubsystem adjustment

Esqatomsmashers-Students committed:
Works: SpecimenArmSubsystem adjustment

Esqatomsmashers-Students committed:
To be tested: BlinkIn colour

Commit:
To be tested: LeftSampleAuto debugging

Esqatomsmashers-Students committed:
Added new opmode and tested all of the limit switch code

 - Tested all limit switch code -> Works
 - Added opmode so one person can control the drive base and sample arm (not the specimen arm)

Esqatomsmashers-Students committed:
Merged drivebase-developmental into mossland-developmental

Drivebase-developmental had some things that will be wanted in mossland-developmental.

Esqatomsmashers-Students committed:
Disabling and renaming OpModes

Some weird things show up but it was just Flotsam emoed (he is teenager)

Esqatomsmashers-Students committed:
Corrected package name

Esqatomsmashers-Students committed:
Corrected package name

Esqatomsmashers-Students committed:
Works: RightSpecimenAuto with 3 spec

Esqatomsmashers-Students committed:
Merged 'drivebase-developmental' into 'mossland-developmental'

Did this so that Bosco's new code could be incorporated into this branch.

Esqatomsmashers-Students committed:
Works: adjusting SpecimenArm position

Esqatomsmashers-Students committed:
Not work: tried to fix drier enhancements

Esqatomsmashers-Students committed:
fixing opMode name

Esqatomsmashers-Students committed:
To be tested: Changes for LeftSampleAuto

Esqatomsmashers-Students committed:
To be tested: Changes for LeftSampleAuto

Esqatomsmashers-Students committed:
To be tested: Changes for claw positions

Esqatomsmashers-Students committed:
To be tested: adjustment of specimen arm

Esqatomsmashers-Students committed:
OnePersonTeleOp control change

Added toggle between specimen and sample mode
Added LED showing them

Esqatomsmashers-Students committed:
To be tested: RRTeleOp

Added it for experimental purpose( not a priority for now)
It can go to where it started in teleOP

Esqatomsmashers-Students committed:
Works: teleOp adjustment from advice

Collin advice: have the wrist in outtake pos when from out-taking to in-taking

Esqatomsmashers-Students committed:
Works: changed commad by advice

Esqatomsmashers-Students committed:
To be tested: faulty LeftSampleAuto adjustment

Esqatomsmashers-Students committed:
To be tested: added remove from submersible

Esqatomsmashers-Students committed:
To be tested: added liftPos in RightSpecimenAuto

Esqatomsmashers-Students committed:
To be tested: sample arm safety feature

Wrist go to intake position before slide retract
and Wrist go to stowed position before put down elbow
(we broke 4 servo in 2 days, darkest week in Robotics)

Esqatomsmashers-Students committed:
Reorganized and renamed programs

Done by Hanson C.

- Refactored programs to correct folders and names
- Disabled unneeded programs
- Added test programs written in OnBotJava

Esqatomsmashers-Students committed:
Reorganized start of programs, and changed accessibility

Done by Hanson C.

- Removed and Added "final" to classes as appropriate
- Reorganized top of programs for consistency

Esqatomsmashers-Students committed:
Reorganized and renamed programs

Done by Hanson C.

- Refactored programs to correct folders and names

Esqatomsmashers-Students committed:
Added IMU angle to teletry for testing

We were experiencing some issues about the field centric drive turning off and wanted to test to see if it was the IMU mal-functioning.

Esqatomsmashers-Students committed:
Works: fixed behaviour of sample arm in Auto

Esqatomsmashers-Students committed:
Works: RightSpecimenAuto

Esqatomsmashers-Students committed:
Works: RightSpecimen Auto

3 specimen, need 3 more seconds to park

Esqatomsmashers-Students committed:
Works: LeftSampleAuto

able to score 2 sample within 20s

Mossk2868 committed:
Added spaces and protection feature in resetEncoders() methods that was overlooked

Added protection when encoders are reset which should make sure the arm or linear slide over extend beyond their limits

Mossk2868 committed:
Renamed package opModes to opmodes, I don't know why it was like this!

Esqatomsmashers-Students committed:
Currently working - all changes added as of first day of competition

Esqatomsmashers-Students committed:
Works: as of 22/2 BC champ
